libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/IrOperationsBuilder.kt

```diff
@@ -787,6 +787,7 @@ internal fun GQLDirective.toDeferBooleanExpression(): BooleanExpression<BTerm>?
 
   val labelArgumentValue = arguments.firstOrNull { it.name == "label" }?.value
   if (labelArgumentValue != null && labelArgumentValue !is GQLStringValue) throw IllegalStateException("Apollo: cannot pass ${labelArgumentValue.toUtf8()} to 'label' argument of 'defer' directive")
+  @Suppress("USELESS_CAST", "KotlinRedundantDiagnosticSuppress")
   val label = (labelArgumentValue as GQLStringValue?)?.value
```

Comment:
With K2, the cast is unnecessary and therefore produces a warning. With 1.9 the cast is needed.